flagutil/flags.go

```diff
@@ -0,0 +1,98 @@
+package flagutil
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+)
+
+func ValidateFlags(flagset *flag.FlagSet, args []string) error {
+	validator := flagValidator{
+		args:    args,
+		flagset: flagset,
+	}
+	for {
+		seen, err := validator.validateNextFlag()
+		if seen {
+			continue
+		}
+		if err == nil {
+			break
+		}
+		return err
+	}
+	return nil
+}
+
+type flagValidator struct {
+	args    []string
+	flagset *flag.FlagSet
+}
+
+// optional interface to indicate boolean flags that can be
+// supplied without "=value" text
+type boolFlag interface {
+	IsBoolFlag() bool
+}
+
+// nolint: gocyclo
+func (v *flagValidator) validateNextFlag() (bool, error) {
+	if len(v.args) == 0 {
+		return false, nil
+	}
+	s := v.args[0]
+	if len(s) < 2 || s[0] != '-' {
```

Comment:
Given these are a little hard to understand, turning them into bool functions could be a nice improvement

Reply:
see https://github.com/atlassian/ctrl/pull/30#discussion_r213725038
I didn't want to alter the logic of parsing the flags...
I'll add this comment to the function.